### ISSUE TITLE
Experimental: exception when using coalesceDefault/unsafeSqlFunctionParens on aliased value

### ISSUE THREAD

**Issue by timds** [bitemyapp/esqueleto#201]
Hi, using the Experimental module, I get this error when using coalesceDefault on the result of a subselect:

```
src/Database/Esqueleto/Internal/Internal.hs:2230:28-68: Non-exhaustive patterns in lambda
```

I don't have time to make a reproduction right now sorry, but it looks like the same problem as that fixed in this pull #189.

I'm using 3.3.3.2, so the error is occurring at the lambda in `unsafeSqlFunctionParens`

**Comment by bitemyapp** [comment]
Shouldn't be occurring if that's your version, it got released prior to 3.3.3.2: https://github.com/bitemyapp/esqueleto/pull/189#issuecomment-647146585

cc @belevy @parsonsmatt Any ideas?

**Comment by belevy** [comment]
Looks like there are a couple more lambdas that assume ERaw they will also fail on compound values. The fix should be fairly easy just annoying that there are two nearly identical functions
